# Laboratorium 1: Wprowadzenie do Pythona i GitHub

## **Cel zadania**
Laboratorium 1 ma na celu:
1. Poznanie zasad organizacji pracy w kursie.
2. Rozwinięcie umiejętności efektywnego korzystania z dokumentacji języka Python.
3. Konfigurację środowiska programistycznego oraz repozytorium na GitHub.
4. Przećwiczenie umiejętności programistycznych w Pythonie.

## **Zakres zadania**
1. Organizacja środowiska pracy.
2. Wyszukiwanie informacji w dokumentacji Pythona.
3. Stworzenie wieloelementowego programu Python.
4. Publikacja wyników na GitHub.

## **Opis programu**
Program napisany w Pythonie wykonuje następujące operacje:
1. Tworzy dwie listy i łączy je za pomocą funkcji `zip()`.
2. Wykorzystuje funkcję `math.sqrt()` z modułu `math` do obliczenia pierwiastka kwadratowego.
3. Obsługuje wyjątek `ZeroDivisionError` za pomocą bloku try-except.
4. Zawiera komentarze z odnośnikami do dokumentacji użytych funkcji i modułów.

## **Użyte moduły, funkcje i wyjątki**

### 1. Funkcja wbudowana: `zip()`
- **Opis**: Funkcja `zip()` łączy elementy z dwóch lub więcej iterowalnych obiektów (np. list) w krotki.
- **Dokumentacja**: [https://docs.python.org/3/library/functions.html#zip](https://docs.python.org/3/library/functions.html#zip)

### 2. Moduł: `math`
- **Opis**: Moduł `math` zawiera funkcje matematyczne, takie jak pierwiastki kwadratowe, logarytmy czy trygonometryczne obliczenia.
- **Dokumentacja**: [https://docs.python.org/3/library/math.html](https://docs.python.org/3/library/math.html)

### 3. Wyjątek: `ZeroDivisionError`
- **Opis**: Wyjątek `ZeroDivisionError` jest podnoszony, gdy próbujemy podzielić liczbę przez zero.
- **Dokumentacja**: [https://docs.python.org/3/library/exceptions.html#ZeroDivisionError](https://docs.python.org/3/library/exceptions.html#ZeroDivisionError)

## **Instrukcje**
1. Kod programu znajduje się w pliku `zadanie_1/zadanie_1.py`.
2. Program wykonuje następujące operacje:
 - Tworzy dwie listy i scala je za pomocą funkcji `zip()`.
 - Wykorzystuje funkcję `math.sqrt()` do obliczenia pierwiastka kwadratowego.
 - Obsługuje wyjątek `ZeroDivisionError` za pomocą bloku try-except.
3. Szczegółowe komentarze w kodzie zawierają odnośniki do dokumentacji.

## **Linki**
- Dokumentacja Pythona: [https://docs.python.org/3/](https://docs.python.org/3/)